src/jit/decomposelongs.cpp

```diff
@@ -1294,6 +1299,144 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 }
 
 //------------------------------------------------------------------------
+// DecomposeRotate: Decompose GT_ROL and GT_ROR with constant shift amounts. We can
+// inspect the rotate amount and decompose to the appropriate node types, generating
+// a shld/shld pattern for GT_ROL, a shrd/shrd pattern for GT_ROR, for most rotate
+// amounts.
+//
+// Arguments:
+//    use - the LIR::Use object for the def that needs to be decomposed.
+//
+// Return Value:
+//    The next node to process.
+//
+GenTree* DecomposeLongs::DecomposeRotate(LIR::Use& use)
+{
+    GenTree* tree       = use.Def();
+    GenTree* gtLong     = tree->gtGetOp1();
+    GenTree* rotateByOp = tree->gtGetOp2();
+
+    genTreeOps oper         = tree->OperGet();
+    genTreeOps rotateByOper = rotateByOp->OperGet();
+
+    assert((oper == GT_ROL) || (oper == GT_ROR));
+    assert((rotateByOper == GT_CNS_INT));
+
```

Comment:
since you don't otherwise use rotateByOper, you could just: assert(rotateByOp->IsCnsIntOrI()) and delete it.

---
In reply to: [84177632](https://github.com/dotnet/coreclr/pull/7724#discussion_r84177632) [](ancestors = 84177632)